### PR TITLE
Refactor microbe editor symmetry from int to enum.

### LINKED ISSUE #998: Change Editor symmetry to be an enum
https://github.com/Revolutionary-Games/Thrive/blob/bf37ff573abcdafe05017db82999c9669449d83c/src/microbe_stage/editor/MicrobeEditor.cs#L124

### PR DESCRIPTION
**Pull request by matthewpaul-us**
In reference to issue #998.

It seems like a pretty easy refactor from the int to the enum. I put in comments for everything, but I wasn't sure about the 6 way symmetry. I couldn't think of a better way to phrase that.

I also tested by going to the editor scene and clicking through the GUI a couple of times to test the changes didn't bust it.

Let me know if I need to change anything. I was having trouble with the formatting checks ruby file. It keeps wanting to make phantom changes to the json files in my local copy. I also had issues with the pre-commit script rejecting the commit silently.

closes #998

### CODE REVIEW

**Comment by hhyyrylainen** [comment]
ok to test

Thanks for the contribution. I haven't tried the pre-commit script on Windows, so if you are on Windows that might be what's the problem (especially regarding line endings).

File: src/microbe_stage/editor/MicrobeEditor.cs [hunk 1]
```diff
@@ -582,54 +600,54 @@ public void RemoveOrganelle()
...
-                break;
-            }
-
-            case 1:
-            {
-                RemoveOrganelleAt(new Hex(q, r));
-
-                if (q != -1 * q || r != r + q)
+            case MicrobeSymmetry.None:
                 {
```
Comment (hhyyrylainen) [inline src/microbe_stage/editor/MicrobeEditor.cs:604]: These curly braces should be indented one level less:
```
        switch (Symmetry)
        {
            case MicrobeSymmetry.None:
            {
                RemoveOrganelleAt(new Hex(q, r));
                break;
            }

            case MicrobeSymmetry.XAxisSymmetry:
            {
```
and so on

Comment (matthewpaul-us) [reply src/microbe_stage/editor/MicrobeEditor.cs]: Thanks for the catch. Honestly, I had never seen a switch statement with curly braces before, so I must have glossed right over it. They aren't idiomatic in C#, so I went ahead and removed them from the switch statement.

**Comment by matthewpaul-us** [comment]
I'm on windows, so that must be it. Not too bad to just undo changes to the json, but it might catch someone if they aren't careful.

**Comment by hhyyrylainen** [comment]
So I had a look at: https://stackoverflow.com/questions/3652408/c-sharp-switch-statement-with-without-curly-brackets-whats-the-difference
and it seems that the reason to use curly braces in a switch would be (almost) the same in C++ and C#, and in C++ I've adopted the style of always using curly braces to avoid the confusing errors about declaring duplicate variables or using variables before initialization.

**Comment by matthewpaul-us** [comment]
I made the switch to use curly braces. There were several other switch statements that didn't have the explicit scope, so I went ahead and changed them for MicrobeEditor. I think that would be a good thing to put in the style guide, as it can differ based on the developer.

**Comment by hhyyrylainen** [comment]
I opened an issue for adding indentation and more CI checks: https://github.com/Revolutionary-Games/Thrive/issues/1285
Also an issue for fixing the formatting script for windows use: https://github.com/Revolutionary-Games/Thrive/issues/1286
And I added on my personal TODO list to clarify the style guide regarding switch statements.

**Pull request merged**